From 7e3ab2d5cdb74618642c9d9f8c9a934e214564cb Mon Sep 17 00:00:00 2001
From: Fabian Vogt <fabian@ritter-vogt.de>
Date: Thu, 16 Sep 2021 18:50:35 +0200
Subject: [PATCH] Fix build on 32bit platforms
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

error: invalid cast from type ‘uintptr_t’ {aka ‘unsigned int’} to type ‘EGLNativeWindowType’ {aka ‘long unsigned int’}
---
 src/plugins/platforms/x11/common/eglonxbackend.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/plugins/platforms/x11/common/eglonxbackend.cpp b/src/plugins/platforms/x11/common/eglonxbackend.cpp
index c8acefe82..32b7c038f 100644
--- a/src/plugins/platforms/x11/common/eglonxbackend.cpp
+++ b/src/plugins/platforms/x11/common/eglonxbackend.cpp
@@ -214,7 +214,7 @@ EGLSurface EglOnXBackend::createSurface(xcb_window_t window)
     }
 
     // Window is 64 bits on a 64-bit architecture whereas xcb_window_t is always 32 bits.
-    uintptr_t nativeWindow = window;
+    Window nativeWindow = window;
 
     EGLSurface surface = EGL_NO_SURFACE;
     if (havePlatformBase()) {
-- 
GitLab

